src/Npgsql/Util/PGUtil.cs

```diff
@@ -19,11 +19,13 @@ static class Statics
     internal static readonly bool LegacyTimestampBehavior;
     internal static readonly bool DisableDateTimeInfinityConversions;
 #endif
+    internal static readonly bool IsNetFramework;
 
     static Statics()
     {
         LegacyTimestampBehavior = AppContext.TryGetSwitch("Npgsql.EnableLegacyTimestampBehavior", out var enabled) && enabled;
         DisableDateTimeInfinityConversions = AppContext.TryGetSwitch("Npgsql.DisableDateTimeInfinityConversions", out enabled) && enabled;
+        IsNetFramework = System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.Contains(".NET Framework");
```

Comment:
We haven't gone into detecting netfx vs. core at runtime, among other things because there doesn't seem to be a good/standard way to do it... Do you think this is important/necessary?

The usages above generally seem to be places where we assumed that netstandard2.0 is being used on netfx (seems reasonable at this point :sweat_smile:), and importantly, if they happen to be on Core, they don't really cause a huge problem either.

Reply:
Well, I'd argue that this is a pretty standard/safe way to detect it but I agree that it is not important.
I Initially did it because I thought that we need to include .NET Standard 2.1 into the conditional compilation (see the latest commit in this PR) and in that case it would have been important. but as it stands it would only affect .NET core < 3.0 which we can probably ignore (OTOH it doesn't hurt neither).